core/shared/src/main/scala/it/unibo/scafi/lib/Gradients.scala

```diff
@@ -5,26 +5,269 @@
 
 package it.unibo.scafi.lib
 
+import scala.concurrent.duration.FiniteDuration
+import scala.math.Numeric.DoubleIsFractional
+import it.unibo.utils.Filters.expFilter
+
 trait StdLib_Gradients {
   self: StandardLibrary.Subcomponent =>
 
   type Metric = ()=>Double
 
+  import Builtins.Bounded
+
+  implicit val idBounded: Bounded[ID]
+
   trait Gradients {
-    self: FieldCalculusSyntax with StandardSensors =>
+    self: FieldCalculusSyntax with StandardSensors with GenericUtils with StateManagement with BlockG =>
```

Comment:
We might consider having `Gradients` extend `GenericUtils with StateManagement with BlockG` instead of using self-type. What do you think? It's unlikely to have different components for those traits, and when mixing `Gradients` in, we don't want users to figure out what other traits need to be mixed in.